addons/hr_expense/models/product_template.py

```diff
@@ -30,6 +31,12 @@ def _auto_init(self):
             )
         return super()._auto_init()
 
-    @api.depends('type')
+    @api.depends('type', 'purchase_ok')
     def _compute_can_be_expensed(self):
-        self.filtered(lambda p: p.type not in ['consu', 'service']).update({'can_be_expensed': False})
+        self.filtered(lambda p: p.type not in ['consu', 'service'] or not p.purchase_ok).update({'can_be_expensed': False})
```

Comment:
`self.filtered(lambda p: p.type not in ['consu', 'service'] or not p.purchase_ok).can_be_expensed = False`

Reply:
why is this diff needed?

Reply:
doing an update with a dictionary containing only one element is useless, but yeah it's not your fault. Feel free to not do it :man_shrugging: